app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+    if request.xhr?
+      render :new, layout: false
+    else
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      if request.xhr?
+        session[:user_id] = @user.id
```

Comment:
`session[:user_id] = @user.id` is in both branches of the if so can move above the if